services/api/src/resources/notification/resolvers.ts

```diff
@@ -711,9 +711,55 @@ export const getAllNotifications: ResolverFn = async (
   args,
   { sqlClientPool, hasPermission }
 ) => {
-  await hasPermission('notification', 'viewAll');
+  let rows;
+
+  if (args.name && args.type) {
+    rows = await query(sqlClientPool, Sql.selectNotificationByNameAndType(args.name, args.type));
+
+    if (rows.length === 0) {
+      throw new Error(`No notification found for ${args.name} & ${args.type}`);
+    }
+
+    if (rows.length > 0) {
+      rows[0].type = args.type;
+    }
+
+    await hasPermission('notification', 'view', {
+      notification: rows[0].id,
+    });
+
+    return rows;
+  }
+
+  if (args.name) {
+    await hasPermission('notification', 'viewAll');
+
+    rows = await Helpers(sqlClientPool).selectAllNotifications(args.name);
 
-  const rows = await Helpers(sqlClientPool).selectAllNotifications();
+    if (rows.length === 0) {
+      throw new Error(`No notifications found for ${args.name}`);
```

Comment:
As this is an admin only query, this error is probably fine to throw

Reply:
Same with the other two errors further down, as they're covered with a `hasPermission('notification', 'viewAll');`